pkg/api/apiloader.go

```diff
@@ -72,23 +71,23 @@ func LoadDefaultContainerServiceProperties() (TypeMeta, *vlabs.Properties) {
 	}
 }
 
-// DeserializeContainerService loads an AKS Cluster API Model, validates it, and returns the unversioned representation
+// DeserializeContainerService loads an AKS Engine Cluster API Model, validates it, and returns the unversioned representation
 func (a *Apiloader) DeserializeContainerService(contents []byte, validate, isUpdate bool, existingContainerService *ContainerService) (*ContainerService, string, error) {
 	m := &TypeMeta{}
 	if err := json.Unmarshal(contents, &m); err != nil {
 		return nil, "", err
 	}
 
 	version := m.APIVersion
-	service, err := a.LoadContainerService(contents, version, validate, isUpdate, existingContainerService)
-	if service == nil || err != nil {
-		if isAgentPoolOnlyClusterJSON(contents) {
-			log.Info("No masterProfile: interpreting API model as agent pool only")
-			service, _, err = a.LoadContainerServiceForAgentPoolOnlyCluster(contents, version, validate, isUpdate, "", existingContainerService)
-			return service, version, err
-		}
+	var cs *ContainerService
+	var err error
+	switch version {
+	case "2017-08-31", "2018-03-31":
```

Comment:
@jackfrancis we should flag this in release notes as a potentially AKS impacting refactor (the commit is test which implies it's only unit tests so it could be easily overlooked).

Reply:
`LoadContainerServiceForAgentPoolOnlyCluster` also supports apvlabs.APIVersion (`vlabs`)

Reply:
AKS doesn't call this function. As far as I can tell the purpose of this function from an AKS perspective is to test AKS-like api models under `pkg/engine/testdata`. AKS never passes a byte array as input (i.e., raw JSON) to aks-engine for processing.

Reply:
Sounds good, thanks for clarifying